### PR TITLE
Fixes #45 - Added an eslint file for logging all the errs in each add-on directory

### LINKED ISSUE #45: Add an npm top level command for running eslint on all add-on folders
The command must eslint the following sub-folders : 
- `firefox`
- `chrome`
- `opera`

PS  : The command must run from the project root

### PR DESCRIPTION
**Pull request by akhilpandey95**
- Removed .eslintrc
- Added .eslint.js

The output looks like this : 
screenshot_2017-01-12_02-47-39

### CODE REVIEW

**Comment by AkhilHector** [comment]
@miketaylr this is how the output looks like : 
screenshot_2017-01-12_02-47-39

**Comment by AkhilHector** [comment]
@miketaylr 2 iquestions : 
Q1.) Should the eslintrc be removed or not ?
Q2.) Should the npm eslint commands in individual add-on directory be updated ?

**Comment by AkhilHector** [comment]
@miketaylr just another thing forgot to say so basically : 
```javascript
logToConsole (someargument, true) // writes the changes to the disk
logToConsole (someargument, false) // doesn't write the changes to the disk
```
PS : I forgot to delete `L13` in `.eslint.js`

File: .eslint.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+const colors = require('colors');
+const eslint = require('eslint').CLIEngine;
+
+var filesToLint = [
+    'chrome/background.js',
+    'firefox/background.js',
+    'firefox-mobile/background.js',
+    'opera/background.js',
+    'test.js'
```
Comment (miketaylr) [inline .eslint.js]: (just a reminder to remove this, you already commented as much :))

File: .eslint.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    globals: [
+        "chrome: true",
+        "content: true",
+        "Components: true",
+        "XPCOMUtils: true",
+        "CustomizableUI: true",
+        "Services: true",
+        "addMessageListener: true",
+        "removeMessageListener: true",
+        "sendAsyncMessage: true"
```
Comment (miketaylr) [inline .eslint.js]: In https://github.com/webcompat/webcompat-reporter-extensions/pull/63, we're going to remove the "firefox-bootstrap" directory, so we won't need any of these globals I think. I'll merge that, and maybe you can double check after rebasing?

File: .eslint.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        "space-before-function-paren": [2, "never"],
+        "space-infix-ops": 2,
+        "space-unary-ops": 2
+    }
+});
+
+var lintReport = lintConfig.executeOnFiles(filesToLint);
+
+function logToConsole(reporFile, saveChanges) {
+    for(let i = 0; i < reporFile.results.length; ++i) {
```
Comment (miketaylr) [inline .eslint.js]: Is it possible to lint this file as well? If not, we can add a space here? `for (let...`

Comment (miketaylr) [inline .eslint.js]: nit: `reportFile`?

File: .eslint.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        process.stdout.write(` ${"Errors: ".green} ${reporFile.results[i].errorCount.toString().bold.magenta}${","}\n`);
+        process.stdout.write(` ${"Warnings: ".green} ${reporFile.results[i].warningCount.toString().bold.blue}${","}\n`);
+        for(let j = 0; j < reporFile.results[i].errorCount; ++j) {
+            process.stdout.write(` ${"Error Message: ".green}${j.toString().green} ${reporFile.results[i].messages[j].message.bold.yellow}${","}\n`);
+            process.stdout.write(` ${"Line, Column: ".green} ${"["}${reporFile.results[i].messages[j].line.toString().underline.cyan}${","}${reporFile.results[i].messages[j].column.toString().underline.cyan}${"]"}\n`);
+        }
+        process.stdout.write(`${"]"}\n`);
+    }
+
+    if(saveChanges) {
```
Comment (miketaylr) [inline .eslint.js]: nit: space after if: `if (`

**Comment by AkhilHector** [comment]
@miketaylr noted all the changes will notify you once i am done.

**Comment by AkhilHector** [comment]
@miketaylr  I have made all the requested changes, here are some of things which you might be interested to know : 

### If you type 
```shell
npm run eslint
```

w1

### If you type
```shell
npm run eslint --info
```

w2

### If you type
```shell
npm run eslint --writetodisk
```

w3

### If you type
```shell
npm run eslint --writetodisk somerandomstring
```

w4

### if you type
```shell
npm run eslint --info --writetodisk 
```

w5

**Comment by miketaylr** [comment]
Thanks! A few thoughts:

> npm run eslint --writetodisk

Why not just make that a boolean flag? Meaning, if you see the flag it means yes, and if you don't see it it means no? Also, `--save` would be less to write.

Also, it seems like we're doing something similar to https://github.com/ratherblue/eslint-html-reporter#usage here. Perhaps we could either use that, or use it's usage patterns? Note that it uses `-o filename`, which is a common unix idiom.

By convention, eslint takes a `-f` custom formatter argument: http://eslint.org/docs/developer-guide/working-with-custom-formatters. 

That said, I probably won't use the save functionality, so I won't block merging on that, but if you'd like to go that route we can clean up the CLI API at least.

**Comment by AkhilHector** [comment]
@miketaylr i have implemented the changes, sorry for taking this much time.

**Comment by miketaylr** [comment]
@AkhilHector when I run `npm run eslint`, I get the following errors:

```
> webcompat-reporter-extensions@ eslint /Users/miket/dev/compat/addons/webcompat-reporter
> node .eslint.js

[
 Filepath:  /Users/miket/dev/compat/addons/webcompat-reporter/chrome/background.js,
 Errors:  1,
 Warnings:  0,
 Error Message: 0 Parsing error: Unexpected character '`',
 Line, Column:  [21,9]
]
[
 Filepath:  /Users/miket/dev/compat/addons/webcompat-reporter/firefox/background.js,
 Errors:  1,
 Warnings:  0,
 Error Message: 0 Parsing error: Unexpected character '`',
 Line, Column:  [20,9]
]
[
 Filepath:  /Users/miket/dev/compat/addons/webcompat-reporter/opera/background.js,
 Errors:  1,
 Warnings:  0,
 Error Message: 0 Parsing error: Unexpected character '`',
 Line, Column:  [21,9]
]
[
 Filepath:  /Users/miket/dev/compat/addons/webcompat-reporter/.eslint.js,
 Errors:  0,
 Warnings:  1,
]
 Changes saved to the disk: false
```

Any idea why it's giving an error for a backtick? That's valid in ES6...

**Comment by AkhilHector** [comment]
@miketaylr looks strange i tried pinging you on IRC to get this problem resolved, honestly i dont understand as to why this error is popping up

**Comment by miketaylr** [comment]
> honestly i dont understand as to why this error is popping up

Hm, we should probably do a little research to understand, and have a follow-up commit to fix the issues. 

I'm hesitant to merge in the code if eslint is saying there are errors.

**Comment by AkhilHector** [comment]
@miketaylr agreed, let me dig in and look for the reason behind this, give me a day's time

**Comment by miketaylr** [comment]
Let's close this, I don't think @akhilpandey95 is working on this right now. :)

**Comment by akhilpandey95** [comment]
Im sorry @miketaylr. I honestly wanted to help, but i am just occupied with other stuff. Hope i could be helpful to the community some other time. ✌️

**Comment by miketaylr** [comment]
No worries. Life gets busy. :)

**Pull request closed without merging**